rotkehlchen/api/v1/schemas.py

```diff
@@ -943,14 +947,67 @@ def make_history_base_entry(
                 data: dict[str, Any],
                 **_kwargs: Any,
         ) -> dict[str, Any]:
-            return {'event': EthWithdrawalEvent(**data)}
+            return {'main_event': EthWithdrawalEvent(**data), 'related_events': None}
+
+    class CreateAssetMovementEventSchema(BaseSchema):
+        event_type = SerializableEnumField(
+            enum_class=HistoryEventType,
+            allow_only=[HistoryEventType.DEPOSIT, HistoryEventType.WITHDRAWAL],
+            required=True,
+        )
+        fee = FeeField(load_default=None)
+        location = LocationField(required=True)
+        unique_id = fields.String(required=False, load_default=None)
+        event_identifier = fields.String(required=False, load_default=None)
+        asset = AssetField(required=True, expected_type=Asset, form_with_incomplete_data=True)
+        fee_asset = AssetField(load_default=None, required=False, expected_type=Asset, form_with_incomplete_data=True)  # noqa: E501
+
+        @post_load
+        def make_history_base_entry(
+                self,
+                data: dict[str, Any],
+                **_kwargs: Any,
+        ) -> dict[str, Any]:
+            if (fee := data['fee']) is not None and data['fee_asset'] is None:
```

Comment:
maybe you want something like

`(fee := data['fee']) is None ^ data['fee_asset'] is None`

using the XOR operator here. You want both of them or none

Reply:
right below you check again `if fee is not None and fee > ZERO:` so maybe can be combined properly.